src/App.tsx

```diff
@@ -1,26 +1,112 @@
-/* eslint-disable max-len */
-/* eslint-disable jsx-a11y/control-has-associated-label */
-import React from 'react';
-import { UserWarning } from './UserWarning';
+import React, { useState, useEffect, useCallback } from 'react';
+import * as todosService from './api/todos';
 
-const USER_ID = 0;
+import { ErrorMessages } from './components/ErrorsMessage';
+import TodoList from './components/TodoList';
+import { Footer } from './components/Footer';
+import Header from './components/Header';
+
+import { Todo } from './types/Todo';
+import { Field } from './types/Field';
+import { filterByTodos, preparedTodos } from './service/service';
 
 export const App: React.FC = () => {
-  if (!USER_ID) {
-    return <UserWarning />;
-  }
+  const [todos, setTodos] = useState<Todo[]>([]);
+  const [field, setField] = useState<Field>(Field.ALL);
+  const [tempTodo, setTempTodo] = useState<Todo | null>(null);
+
+  const [isDeleted, setIsDeleted] = useState(false);
+  const [todosForDelete, setTodosForDelete] = useState<Todo[]>([]);
+  const [errorMessage, setErrorMessage] = useState('');
+
+  useEffect(() => {
+    todosService
+      .getTodos()
+      .then(setTodos)
+      .catch(() => setErrorMessage('Unable to load todos'));
+  }, []);
+
+  const todosCounter = filterByTodos(todos);
+
+  useEffect(() => {
+    const timerId = window.setTimeout(() => setErrorMessage(''), 3000);
+
+    return () => clearTimeout(timerId);
+  }, [errorMessage]);
+
+  const filteredTodos = preparedTodos(todos, field);
+
+  const changeComplete = useCallback(
+    (todoChanged?: Todo) => {
+      const changedTodos = todos.map(todo => {
+        if (todo.id !== todoChanged?.id) {
+          return todo;
+        } else {
+          return { ...todo, completed: !todo.completed };
+        }
+      });
+
+      setTodos(changedTodos);
+    },
+    [todos],
+  );
+
+  const addTempTodo = useCallback((temporaryTodo: Todo | null) => {
+    if (temporaryTodo) {
+      setTempTodo({ ...temporaryTodo, id: 0 });
+    } else {
+      setTempTodo(null);
+    }
```

Comment:
```suggestion
      setTempTodo(temporaryTodo ? { ...temporaryTodo, id: 0 } : null);
```

can be a bit simplify